File: practica1/P1S3/C++/Filtro2.cpp
#pragma once
#include "Filtro.cpp"
#include "Comida.cpp"
#include <iostream>
using namespace std;

class Filtro2 : public Filtro
{   
    public:
    Filtro2(){}

    Filtro2(Comida *c)
    {
        comida = c;
    }

    void ejecutar()
    {
        cout << "Se ha aplicado un descuento del 50%" << endl;
        comida->setPrecio(comida->getPrecio() * 0.5);
    }

    void setComida(Comida *c)
    {
        comida = c;
    }
    
};

File: practica1/P1S3/Java/src/practica1dss3/Filtro2.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss3;

/**
 *
 * @author USUARIO
 */
public class Filtro2 implements Filtro{
    private Comida comida;
    
    public Filtro2(){
        
    }

    @Override
    public void ejecutar() {
        System.out.println("Se ha aplicado un descuento del 50%");
        comida.setPrecio(comida.getPrecio() * 0.5);
    }

    @Override
    public void setComida(Comida c) {
        comida = c;
    }
}
